### PR TITLE
Read test stack image address replacement from external file

### PR DESCRIPTION
**Pull request by dmytro-ndp**
This PR involves provider of Map[**stack-image-address-prefix -> new-stack-image-address**].
Map should be placed in json file "src/test/resources/conf/stack-replacement.json" with content like the follow: 
```
{
 "registry.access.redhat.com/codeready-workspaces-beta/stacks-java-rhel8": "quay.io/crw/stacks-java-rhel8:1.1-1",
 "registry.access.redhat.com/codeready-workspaces/stacks-cpp": "quay.io/crw/stacks-cpp:1.1-9",
 "registry.access.redhat.com/codeready-workspaces/stacks-dotnet": "quay.io/crw/stacks-dotnet:1.1-5",
 "registry.access.redhat.com/codeready-workspaces/stacks-golang": "quay.io/crw/stacks-golang:1.1-1",
 "registry.access.redhat.com/codeready-workspaces/stacks-java": "quay.io/crw/stacks-java:1.1-11",
 "registry.access.redhat.com/codeready-workspaces/stacks-node": "quay.io/crw/stacks-node:1.1-8",
 "registry.access.redhat.com/codeready-workspaces/stacks-php": "quay.io/crw/stacks-php:1.1-1",
 "registry.access.redhat.com/codeready-workspaces/stacks-python": "quay.io/crw/stacks-python:1.1-3"
}
```
Example of map to quay.io latest tags could be found in **test/codeready-test-e2e/src/test/resources/conf/stack-replacement-quay-latest.json** file.

Signed-off-by: Dmytro Nochevnov <dnochevn@redhat.com>

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by artaleks9** [comment]
ci-test

**Comment by dmytro-ndp** [comment]
@nickboldt: thank you for valuable feedback.
Please, review fixup.

File: test/codeready-test-e2e/src/main/java/com/redhat/codeready/selenium/core/provider/TestStackAddressReplacementProvider.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/*
+* Copyright (c) 2018 Red Hat, Inc.
```
Comment (nickboldt) [inline test/codeready-test-e2e/src/main/java/com/redhat/codeready/selenium/core/provider/TestStackAddressReplacementProvider.java]: Maybe you mean 2019?

File: test/codeready-test-e2e/src/main/java/com/redhat/codeready/selenium/core/utils/CodeReadyWorkspaceDtoDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*
+* Copyright (c) 2018 Red Hat, Inc.
+
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
```
Comment (nickboldt) [inline test/codeready-test-e2e/src/main/java/com/redhat/codeready/selenium/core/utils/CodeReadyWorkspaceDtoDeserializer.java]: Please fix copyrights to use current year and EPL v2.0.

**Comment by dmytro-ndp** [comment]
Thank you for comments. Nick!
I would like to fix the year and set EPL v2.0. by separate PR.

**Comment by nickboldt** [comment]
JIRA for EPL 2 (and fixing copyright dates if applicable) is https://issues.jboss.org/browse/CRW-230

**Pull request merged**